astroquery/ipac/irsa/tests/test_most.py

```diff
@@ -0,0 +1,249 @@
+    return mp
+
+
+def test_validation(patch_requests):
+    """Tests MOST parameter validation."""
+    with pytest.raises(ValueError):
+        Most.query_object()
+
+    with pytest.raises(ValueError):
+        Most.query_object(catalog="wise_allsky_4band")
+
+    # make sure no funny business happens with
+    # overwriting of default values
+    Most.query_object(
+        catalog="wise_allsky_4band",
+        obj_name="Victoria"
+    )
+    Most.query_object(
+        catalog="wise_allsky_4band",
+        obj_name="Victoria",
+        input_mode="name_input"
+    )
+
+    with pytest.raises(ValueError):
+        # fails because insufficient orbital parameters specified
+        Most.query_object(
+            catalog="wise_allsky_4band",
+            output_mode="Brief",
+            input_mode="manual_input",
+            obj_type="Asteroid",
+            perih_dist=1.5,
+            eccentricity=0.5
+        )
+
+    Most.query_object(
+        catalog="wise_allsky_4band",
+        output_mode="Brief",
+        input_mode="manual_input",
+        obj_type="Asteroid",
+        semimajor_axis=2.68,
+        eccentricity=0.33,
+    )
+
+    with pytest.raises(ValueError):
+        # Comets require perihel_dist keyword
+        # instead of smimajor_axis
+        Most.query_object(
+            catalog="wise_allsky_4band",
+            output_mode="Brief",
+            input_mode="manual_input",
+            obj_type="Comet",
+            semimajor_axis=2.68,
+            eccentricity=0.33
+        )
+
+    with pytest.raises(ValueError):
+        # object type is case sensitive
+        Most.query_object(
+            catalog="wise_allsky_4band",
+            output_mode="Brief",
+            input_mode="manual_input",
+            obj_type="comet",
+            semimajor_axis=2.68,
+            eccentricity=0.33
+        )
+
+    with pytest.raises(ValueError):
+        # missing mpc_data and obj_type
+        Most.query_object(
+            catalog="wise_allsky_4band",
+            output_mode="Brief",
+            input_mode="mpc_input"
+        )
+
+    Most.query_object(
+        catalog="wise_allsky_4band",
+        output_mode="Brief",
+        input_mode="mpc_input",
+        obj_type="Asteroid",
+        mpc_data="K10N010+2010+08+16.1477+1.494525+0.533798+153.4910+113.2118+12.8762+20100621+17.0+4.0+P/2010+N1+(WISE)+MPC+75712"  # noqa: E501
+    )
+
+
+def test_regular(patch_requests):
+    """Tests the default MOST query returns expected results."""
+    response = Most.query_object(obj_name="Victoria")
+
+    assert "results" in response
+    assert "metadata" in response
+    assert "region" in response
+    assert "fits_tarball" not in response
+    assert "region_tarball" not in response
+
+    results = Table.read(data_path("most_regular_results.tbl"), format="ipac")
+    metadata = Table.read(data_path("most_imgframes_matched_final_table.tbl"), format="ipac")
+    url = REGULAR_BASE_URL + "ds9region/ds9_orbit_path.reg"
```

Comment:
Super nitpick on naming things: maybe call these expected, e.g. expected_result, and the `response` above as `result`, too? As it's not the response, but the final result, e.g. the parsed query respose.